modules/gke-cluster/main.tf

```diff
@@ -140,7 +140,17 @@ resource "google_container_cluster" "cluster" {
   dynamic "cluster_autoscaling" {
     for_each = var.cluster_autoscaling == null ? [] : [""]
     content {
-      enabled = true
+      enabled = var.enable_features.autopilot ? null : true
```

Comment:
I think instead of setting `enabled` when autopilot is request, just disable the whole block (adding the extra check to the outer cluster_autoscaling block). Something like this:
```hcl
dynamic "cluster_autoscaling" {
    for_each = (var.cluster_autoscaling == null || var.enable_features.autopilot) ? [] : [""]
```

Reply:
@juliocc  that would remove the feature this PR adds which is the ability to specify auto_provisioning_defaults, a block that is contained within cluster_autoscaling

Reply:
it will remove it only when autopilot is enabled, so as not to break autopilot AFAIK

Reply:
Not exactly. The problem I'm trying to address here is that we should be able to specify a non-default node service account when autopilot is enabled. When the cluster_autoscaling block is provided enabled is defaulted to true and it fails at the plan stage

`Error: Conflicting configuration arguments
│
│   with module.test_gke_cluster.google_container_cluster.cluster,
│   on .terraform/modules/test_gke_cluster/modules/gke-cluster/main.tf line 17, in resource "google_container_cluster" "cluster":
│   17: resource "google_container_cluster" "cluster" {
│
│ "cluster_autoscaling.0.enabled": conflicts with enable_autopilot`

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#nested_cluster_autoscaling
> (Optional) Whether node auto-provisioning is enabled. Must be supplied for GKE Standard clusters, true is implied for autopilot clusters

Reply:
I remember a provider bug that prevented using a different SA with autopilot, has that been fixed? I can try finding it...

Reply:
I've got a cluster up and running with a custom SA using the change implemented here

Reply:
ahhhh nice!

Reply:
let's approve then. We can always come back if anything breaks